File: _code_examples/kamon-core-examples/src/main/java/kamon/examples/java/UserMetrics.java
package kamon.examples.java;

import kamon.Kamon;
import kamon.metric.instrument.Counter;
import kamon.metric.instrument.Histogram;

public class UserMetrics {
  public static void main(String[] args) {
    final Kamon kamon = Kamon.create();

    final Histogram myHistogram = kamon.userMetrics().histogram("my-histogram");
    myHistogram.record(42);
    myHistogram.record(43);
    myHistogram.record(44);

    final Counter myCounter = kamon.userMetrics().counter("my-counter");
    myCounter.increment();
    myCounter.increment(17);

    kamon.shutdown();
  }
}

File: _code_examples/kamon-core-examples/src/main/scala/kamon/examples/scala/UserMetrics.scala
package kamon.examples.scala

import kamon.Kamon

object UserMetrics extends App {
  val kamon = Kamon()

  val myHistogram = kamon.userMetrics.histogram("my-histogram")
  myHistogram.record(42)
  myHistogram.record(43)
  myHistogram.record(44)

  val myCounter = kamon.userMetrics.counter("my-counter")
  myCounter.increment()
  myCounter.increment(17)

  kamon.shutdown()
}
